jest.setup.js

```diff
@@ -1,10 +1,34 @@
+import { readFileSync } from 'fs';
+import { join } from 'path';
+
 // TODO [#339]: replace third-party libraries with NPM modules
 import jQuery from './_dprhtml/js/external/jquery-3.4.1.min';
 import knockout from './_dprhtml/js/external/ajax/libs/knockout/3.5.0/knockout-min';
 
+window.$ = jQuery;
 window.jQuery = jQuery;
 window.ko = knockout;
 
 window.DPR_G = require('./_dprhtml/js/globalObject');
 window.DPR_PAL = require('./_dprhtml/js/dpr_pal');
+window.DPR_translitCore_mod = require('./_dprhtml/js/translitCore');
+window.DPR_translit_mod = require('./_dprhtml/js/translit');
+window.DPR1_chrome_mod = require('./_dprhtml/js/chrome').DPR1_chrome_mod;
+window.DPR1_format_mod = require('./_dprhtml/js/format');
+window.DPR_Chrome = require('./_dprhtml/js/chrome').DPR_Chrome;
+window.DPR_navigation_mod = require('./_dprhtml/js/navigation');
+window.DPR_navigation_common_mod = require('./_dprhtml/js/navigation_common');
 window.DPR_prefload_mod = require('./_dprhtml/js/prefload');
+window.DPR_search_mod = require('./_dprhtml/js/web/search');
+window.DPR_search_utils_mod = require('./_dprhtml/js/search_utils');
+window.DPR_sortaz_mod = require('./_dprhtml/js/sortaz');
+window.XML_Load = require('./_dprhtml/js/xml_load').XML_Load;
+
+window.focus = jest.fn();
+
+window.XML_Load.xhrGetAsync = ({ url }, transformResponse) => {
+  const xmlFilepath = join(__dirname, url);
+  const xmlString = readFileSync(xmlFilepath).toString();
+  const responseXML = new DOMParser().parseFromString(xmlString, 'text/xml');
+  return transformResponse({ responseXML });
+};
```

Comment:
Overwriting `XML_Load.xhrGetAsync` allows us to run `searchTipitaka` as though it were a real search, including asynchronously loading XML files. This level of integration testing can be useful in increasing our confidence that the code is working correctly since the path of execution in our tests is quite similar to that of the browser.

Reply:
@parthopdas I had an alternative idea. There is a library called [Mock Service Worker](https://github.com/mswjs/msw) that would allow us to intercept network requests made in the test environment. We could have `msw` listen to requests to `localhost` and fulfill the request via an async `readFile` operation.

As an aside, I've used `msw` for a bunch of projects and it works fantastically (even Kent C Dodds, the author of `@testing-library`, [recommends msw](https://kentcdodds.com/blog/stop-mocking-fetch#then-i-discovered-msw) as the preferred approach for mocking network calls in frontend tests).

Reply:
Just to follow up on this: I've worked with the maintainer of msw to merge a fix for this issue, which should be included in the next release of msw/interceptors. The msw release will following shortly after. See:
- https://github.com/mswjs/msw/issues/715
- https://github.com/mswjs/interceptors/pull/117